app/controllers/Application.scala

```diff
@@ -60,10 +60,15 @@ final class Application @Inject()(data: DataHelper,
       val actions = this.service.getSchema(classOf[ProjectSchema])
 
       val canHideProjects = this.users.current.isDefined && (this.users.current.get can HideProjects in GlobalScope)
-      val visibleFilter: ModelFilter[Project] = if (!canHideProjects)
-        ModelFilter[Project](_.isVisible)
+      var visibleFilter: ModelFilter[Project] = if (!canHideProjects)
+        ModelFilter[Project](_.visibility === VisibilityTypes.Public)
       else
         ModelFilter.Empty
+      if (this.users.current.isDefined) {
+        val currentUser = this.users.current.get
+        visibleFilter = visibleFilter +|| (ModelFilter[Project](_.userId === currentUser.id.get)
+            +&& ((ModelFilter[Project](_.visibility === VisibilityTypes.NeedsChanges) +|| ModelFilter[Project](_.visibility === VisibilityTypes.Draft))))
```

Comment:
Doesn't this leave out projects that the user owns that `NeedsApproval`? I think checking for `ModelFilter[Project](_.visibility =!= VisibilityTypes.SoftDelete)` would be cleaner.